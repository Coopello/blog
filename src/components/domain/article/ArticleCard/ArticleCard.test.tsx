import renderer from "react-test-renderer";

import { ArticleCard } from "./";

describe("ui/ArticleCard", () => {
  it("Snap Shot ArticleCard", () => {
    const component = renderer.create(
      <ArticleCard
        imageUrl="https://placehold.jp/150x150.png"
        nickname="ニックネーム"
        description="XXX Developer"
        tagsInfo={[
          { type: "chip", text: "chip" },
          { type: "frontend", text: "frontend" },
        ]}
        title="タイトル"
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
