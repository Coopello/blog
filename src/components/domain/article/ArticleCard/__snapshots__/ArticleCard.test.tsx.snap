// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/ArticleCard Snap Shot ArticleCard 1`] = `
<div
  className="flex w-[380px] max-w-full flex-col gap-4 rounded-xl border p-6"
>
  <h2
    className="border-b pb-2 text-2xl"
  >
    タイトル
  </h2>
  <div
    className="flex gap-4"
  >
    <img
      alt="プロフィール画像"
      className="rounded-full object-contain"
      data-nimg="1"
      decoding="async"
      height={48}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=https%3A%2F%2Fplacehold.jp%2F150x150.png&w=96&q=75"
      srcSet="/_next/image?url=https%3A%2F%2Fplacehold.jp%2F150x150.png&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fplacehold.jp%2F150x150.png&w=96&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={48}
    />
    <div>
      <h3
        className="text-base"
      >
        ニックネーム
      </h3>
      <p
        className="text-base text-custom-gray"
      >
        XXX Developer
      </p>
    </div>
  </div>
  <ul
    className="flex gap-2"
  >
    <li>
      <span
        className="rounded-sm px-3 py-1 text-sm"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          {
            "backgroundColor": "white",
            "color": "#222222",
          }
        }
      >
        #
        chip
      </span>
    </li>
    <li>
      <span
        className="rounded-sm px-3 py-1 text-sm"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          {
            "backgroundColor": "white",
            "color": "#FF7971",
          }
        }
      >
        #
        frontend
      </span>
    </li>
  </ul>
</div>
`;
